### ISSUE TITLE
CI reported detekt-formatting issue that wasn't found locally

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#6111]
## Expected Behavior
Local `gradlew detekt*` finds the same problems as CI

## Observed Behavior
Check out https://github.com/detekt/detekt/commit/e2a3af064762518976391edac9de01e590f6f8b4 from https://github.com/detekt/detekt/pull/5981 and run detekt on it offline -> no findings.

https://github.com/detekt/detekt/security/code-scanning/7293 -> found by CI.

## Steps to Reproduce
In `detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenComment.kt` replace method with
```kotlin
    private fun getErrorMessage(value: String): String =
        @Suppress("DEPRECATION")
        customMessage.takeUnless { it.isEmpty() } ?: String.format("DEFAULT_ERROR_MESSAGE__%s__NO_REASON", value)
```
(It doesn't make much sense, but hopefully it should repro if this is pushed in a PR.)

## Context
> Hi, @TWiStErRob do you know why this didn't get caught when I ran the `./gradlew detektMain detektTest detektTestFixtures apiDump generateDocumentation compileKotlin compileTestKotlin compileTestFixturesKotlin -PwarningsAsErrors=true --parallel` command locally? Running the command successfully passed on my local

_Originally posted by @atulgpt in https://github.com/detekt/detekt/pull/5981#discussion_r1195748650_

## Your Environment

* Version of detekt used: 1.23 RCx
* Version of Gradle used (if applicable): 8.0.2
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: Linux and Windows
* Link to your project (if it's a public repository): https://github.com/atulgpt/detekt/tree/e2a3af064762518976391edac9de01e590f6f8b4
